### PR TITLE
Remove DNS hacks in dns listener

### PR DESCRIPTION
**Pull request by rshriram**
The bug in Envoy has been fixed.

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: tests/integration/iop-integration-test-defaults.yaml [hunk 1]
```diff
@@ -4,9 +4,9 @@ kind: IstioOperator
 spec:
   meshConfig:
     accessLogFile: "/dev/stdout"
-    #defaultConfig:
-    #  proxyMetadata:
-    #    ISTIO_META_DNS_CAPTURE: "ALL"
+    defaultConfig:
```
Comment (howardjohn) [inline tests/integration/iop-integration-test-defaults.yaml:7]: other changes LGTM, is this expected though?

File: pilot/pkg/networking/core/v1alpha3/dns_listener.go [hunk 2]
```diff
@@ -73,7 +73,7 @@ func (configgen *ConfigGeneratorImpl) buildSidecarDNSListener(node *model.Proxy,
 		ClientConfig: &dnsfilter.DnsFilterConfig_ClientContextConfig{
 			ResolverTimeout: ptypes.DurationProto(resolverTimeout),
 			// no upstream resolves. Envoy will use the ambient ones
-			MaxPendingLookups: 64, // arbitrary
+			MaxPendingLookups: 128, // arbitrary
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/dns_listener.go:76]: We can probably go up to 256 if we want -that is what Envoy docs and integration tests are defaulting it to.

File: tests/integration/pilot/vm/vm_test.go [hunk 3]
```diff
@@ -79,8 +79,8 @@ spec:
 `)
 			ports := []echo.Port{
 				{
-					Name:         "http",
-					Protocol:     protocol.HTTP,
+					Name:         "grpc",
+					Protocol:     protocol.GRPC,
```
Comment (hzxuzhonghu) [inline tests/integration/pilot/vm/vm_test.go:83]: why are protocol related?

File: tests/integration/iop-integration-test-defaults.yaml (hunk 1 above)
Comment (rshriram) [reply tests/integration/iop-integration-test-defaults.yaml:7]: yes, trying to get dns work in our tests on k8s. something is off - only vm works but not pods - esp for grpc

**Pull request merged**